docs/how-to-guides/upgrade-openebs.md

```diff
@@ -0,0 +1,279 @@
+# Upgrading and migrating OpenEBS data plane components
+
+## Contents
+
+* [Introduction](#introduction)
+* [Prerequisites](#prerequisites)
+* [Step 1: Upgrade OpenEBS control plane components](#step-1-upgrade-openebs-control-plane-components)
+* [Step 2: Upgrade OpenEBS data plane components](#step-2-upgrade-openebs-data-plane-components)
+	* [Upgrade cStor pools](#upgrade-cstor-pools)
+	* [Upgrade cStor volumes](#upgrade-cstor-volumes)
+* [Step 3: Verify](#step-3-verify)
+* [Summary](#summary)
+* [Additional resources](#additional-resources)
+
+## Introduction
+
+This guide provides steps to upgrade OpenEBS dataplane components from `v1.12.0`
+to `v2.2.0` and migrate from cStor Pools/Volumes to latest CSI based
+provisioning (CSPC Pools/Volumes).
+
+## Prerequisites
+
+To upgrade the OpenEBS data plane components, we need the following:
+
+* Lokomotive `v0.6.0`
+    ```bash
+    lokoctl version
+    v0.6.0
+    ```
+
+* A Lokomotive cluster accessible via `kubectl` deployed on a
+  [Equinix metal](../configuration-reference/platforms/packet.md) or
+  [Bare metal](../configuration-reference/platforms/baremetal.md).
+
+* OpenEBS `v1.12.0` installed. You can check if OpenEBS is indeed in expected
+    version:
+    ```bash
+    $ kubectl get pods -n openebs -n openebs.io/version=1.12.0
+    ```
+
+* Highly recommended to schedule a downtime for the applications consuming the
+  OpenEBS PV and make sure to take a backup of the data before starting the
+  below upgrade procedure. Lokomotive provides
+  [Velero](../configuration-reference/components/velero.md) component for backup
+  and restore.
+
+* Ensure the cluster and OpenEBS volumes are is in healthy state before proceeding.
+
+## Steps
+
+### Step 1: Upgrade OpenEBS control plane components
+
+Lokomotive provides an easy way of upgrading the OpenEBS control plane.
+
+Execute the following command to upgrade OpenEBS control plane:
+
+```bash
+lokoctl component apply openebs-operator
+```
+
+Doing so, terminates the OpenEBS resources associated with `v1.12.0` and new
+resources are created with `v2.2.0`.
+
+Verify all the pods are in `Running` state before proceeding:
+
+```bash
+$ kubectl get pods -n openebs -n openebs.io/version=2.2.0
+```
+
+### Step 2: Upgrade OpenEBS data plane components
+
+OpenEBS control plane and data plane components work independently. Even if the
+control plane components are upgraded to `v2.2.0`, the data plane components
+continue to work with the older version `v1.12.0`.
+
+#### Upgrade cStor pools
+
+Get existing `StoragePoolClaims` by executing:
+
+```bash
+kubectl get spc
+NAME                           AGE
+cstor-pool-openebs-replica-3   2d21h26m
+```
+
+Create a Job to upgrade the existing cStor pools:
+
+```bash
+OPENEBS220=2.2.0
```

Comment:
I doubt @surajssd will like `OPENEBS220` variable name, I don't think this is what he meant :smile: I think we should have something like `OLD_VERSION` and `NEW_VERSION` or something.

Reply:
`OPENEBS_NEW_VERSION` and `OPENEBS_OLD_VERSION`